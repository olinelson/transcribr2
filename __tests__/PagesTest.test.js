import React from 'react'
import renderer from 'react-test-renderer'

import FourOFour from '../src/pages/404'
import About from '../src/pages/about'
import Forgot from '../src/pages/Forgot'

beforeAll(() => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn()
    }))
  })
})

describe('404 page', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FourOFour />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

// to do... spoof graphql images

// describe('about page', () => {
//   it('renders correctly', () => {
//     const tree = renderer
//       .create(<About />)
//       .toJSON()
//     expect(tree).toMatchSnapshot()
//   })
// })

// describe('forgot password page', () => {
//   it('renders correctly', () => {
//     const tree = renderer
//       .create(<Forgot />)
//       .toJSON()
//     expect(tree).toMatchSnapshot()
//   })
// })
