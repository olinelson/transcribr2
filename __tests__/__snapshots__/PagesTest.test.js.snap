// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page renders correctly 1`] = `<fourOFour />`;
